### ISSUE TITLE
translations: Some strings are not picked up

### ISSUE THREAD

**Issue by marusak** [cockpit-project/cockpit#12835]
This happens with `.js` and `.jsx` files (at least where I noticed it) when string is in single quotes (\`).
This is due to fact, that we use `--language=C` in `xgettext` in `po/Makefile`. There is also supported language `--language=Javascript`, but there is problem that `xgettext` cannot work with files, that contain self enclosing tags. I [reported](https://savannah.gnu.org/bugs/?56848) this issue.

Only solution that would work now is to enforce using double quotes.

How to see if this is working:
`make update/po` and then check in `po/cockpit.pot` if there is for example `September` from `pkg/lib/journal.js`.

Related: https://bugzilla.redhat.com/show_bug.cgi?id=1752862

**Comment by martinpitt** [comment]
@marusak: Is this really about self-closing tags, or really just about the single quotes? It's well known that single quotes don't work with gettext, and the `js-translatable-strings` test in tools/test-static-code tries to enforce this (and it has saved us a couple of times). However, it wouldn't see ` quotes (backticks), as you have in your description. Was this just a typo and you really meant ' for that particular affected string?

either way, we should add ` to the list of "bad" string delimiters for i18n.

**Comment by martinpitt** [comment]
I tried this, and it didn't find anything:

```diff
--- a/tools/test-static-code
+++ b/tools/test-static-code
@@ -49,7 +49,7 @@ fi
 # wrongly marked translatable strings
 #
 
-if out=$(find src/ pkg/ -name '*.js' -o -name '*.jsx' | xargs grep "_('"); then
+if out=$(find src/ pkg/ -name '*.js' -o -name '*.jsx' | xargs grep "_(['\`]"); then
     echo 'ERROR: translatable strings must be marked with _("")' >&2
     echo "$out" >&2
     echo "not ok 3 js-translatable-strings"

```

I post this as a PR, but this isn't the cause.

**Comment by marusak** [comment]
> Is this really about self-closing tags, or really just about the single quotes?

It is about single quotes when `xgettext` is ran with `--language=C` and it becomes problem with self-enclosing tags when `xgettext` is ran with `--language=Javascript`.

> Was this just a typo and you really meant ' for that particular affected string?

It was typo, I meant ', sorry.

**Comment by martinpitt** [comment]
So we have multiple different issues here:

 * The original bugzilla which means we must use moment to show these dates on the journal, instead of trying to implement our own.
 * Using (and not checking `ngettext('...')` in a couple of places.
 * Not checking quote style for multi-arg ngettext.

**Comment by marusak** [comment]
> - The original bugzilla which means we must use moment to show these dates on the journal, instead of trying to implement our own

The months are not picked up, because `gettext` (not `Ngettext`) has two arguemnts - and first one is in double quotes and the second one (the real string) is in single quotes. But moment would (probably) work as well.

> - Using (and not checking ngettext('...') in a couple of places.
>  - Not checking quote style for multi-arg ngettext.

One of them should be `gettext`, right?

**Comment by martinpitt** [comment]
> Not checking quote style for multi-arg gettext

FTR, this isn't possible to do with just grep, as that's context sensitive. I went through

    find src/ pkg/ -name '*.js' -o -name '*.jsx' | xargs grep -E "(gettext|_)\([^)]*['\`]"

and found one more case, I'll fix it in my PR.

**Comment by martinpitt** [comment]
Fixed in #12841.
